### DISCUSSION TITLE
Hosting Pocketbase from an AZ VM.

### DISCUSSION THREAD

**Discussion by untold-titan** [pocketbase/pocketbase#452] [Q&A] [answered]
Hi! 
I'm trying to host my pocketbase server on an AZ VM running ubuntu, and when I try to run it, it says this: 
```
sudo ./pocketbase serve --https="20.171.88.149:443"
> Server started at: https://20.171.88.149:443
  - REST API: https://20.171.88.149:443/api/
  - Admin UI: https://20.171.88.149:443/_/
2022/09/14 23:45:29 listen tcp 20.171.88.149:443: bind: cannot assign requested address
```
If anyone knows why pocketbase can't seem to access this address, please let me know. 

This is a brand new install of ubuntu btw, the only thing I've installed is pocketbase.

**Comment by ganigeorgiev** [comment, marked as answer]
When PocketBase is started with the `--https` flag it will generate a Let's encrypt certificate for you, but Let's encrypt currently doesn't support issuing certificates for bare ips (https://community.letsencrypt.org/t/certificates-for-ip-addresses/158093/5).

Or in other words - you need a valid domain name if you want to use the auto SSL feature.

You can always start PocketBase with the `--http` flag and if you need SSL put [PocketBase behind a reverse proxy](https://pocketbase.io/docs/going-to-production/#deploy-behind-reverse-proxy) in combination with certbot and bring your own certificate.

**Reply by untold-titan** [reply]
Oh, ok. Makes sense, I have an ssl cert I can bind to my domain, cause I have a domain. And then I'll just try to use that when running pocketbase.

It also gives this same error if I use it with the `--http` flag.
I have also tried to use the domain on it as well. 
The domain is cataclysmpocket.tech

**Reply by ganigeorgiev** [reply]
> I have an ssl cert I can bind to my domain, cause I have a domain.

You don't have to use a custom SSL if you have a domain. You need to just configure your dns so that the domain points to the vm and then you'll be able to start PocketBase by just running:
```
./pocketbase serve --http="cataclysmpocket.tech:80" --https="cataclysmpocket.tech:443"
```

---

> It also gives this same error if I use it with the --http flag.

That's strange. I don't have experience with the Azure cloud vms (_assuming that this is what you mean by "AZ VM"_) and I don't know whether there are some specical configurations that you need to do, but usually if you can't bind directly to the domain, or the IP that resolves to it, that means that the server is probably missing a loopback address (you could check your `/etc/hosts`).

You can also try binding to the designated `0.0.0.0` address, which means - bind to all available IPv4 interfaces on the vm (nginx and other proxies usually bind to this address).

**Reply by untold-titan** [reply]
TYSM! I used the reverse proxy, and it instantly started working. Thanks!

**Reply by Hammaad-M** [reply]
@untold-titan Can you please share what you did? I'm having the same "cannot assign requested address error" when running the serve cmd on an Azure VM. Tried http and https, with <public_ip>:80 as the location. 
Got it to run with 0.0.0.0:80 but can't access the admin dashboard at <public_ip>:80. 

I have also run the setcap cmd to give the executable permission to bind. Running on Linux ubuntu 20.04.

Any help would be really appreciated.
